core/src/main/java/io/lionweb/lioncore/java/serialization/DeserializationStatus.java

```diff
@@ -0,0 +1,74 @@
+package io.lionweb.lioncore.java.serialization;
+
+import io.lionweb.lioncore.java.model.impl.ProxyNode;
+import io.lionweb.lioncore.java.serialization.data.SerializedClassifierInstance;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+
+/**
+ * This class is used to track the status of the deserializationg process, and in particular sorting
+ * operations. It is also used to track the proxies created while sorting this particular set of
+ * nodes, as the proxies created in this process will need to be returned together with the sorted
+ * list of unserialized nodes.
+ */
+class DeserializationStatus {
+  List<SerializedClassifierInstance> sortedList;
```

Comment:
I think all of these fields can be `final`?

Reply:
Making all these fields final in follow up commit